src/modules/hg_branch.rs

```diff
@@ -0,0 +1,90 @@
+use std::process::Command;
+use unicode_segmentation::UnicodeSegmentation;
+
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::hg_branch::HgBranchConfig;
+
+/// Creates a module with the Hg bookmark or branch in the current directory
+///
+/// Will display the bookmark or branch name if the current directory is an hg repo
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_hg_repo = context
+        .try_begin_scan()?
+        .set_files(&[".hgignore"])
+        .set_folders(&[".hg"])
+        .is_match();
+
+    if !is_hg_repo {
+        return None;
+    }
+
+    let mut module = context.new_module("hg_branch");
+    let config = HgBranchConfig::try_load(module.config);
+    module.set_style(config.style);
+
+    module.get_prefix().set_value("on ");
+
+    let truncation_symbol = get_graphemes(config.truncation_symbol, 1);
+    module.create_segment("symbol", &config.symbol);
+
+    // TODO: Once error handling is implemented, warn the user if their config
+    // truncation length is nonsensical
+    let len = if config.truncation_length <= 0 {
+        log::warn!(
+            "\"truncation_length\" should be a positive value, found {}",
+            config.truncation_length
+        );
+        std::usize::MAX
+    } else {
+        config.truncation_length as usize
+    };
+
+    let mut branch_name = match get_hg_log_template(context, "{activebookmark}") {
+        None => get_hg_log_template(context, "{branch}").unwrap(),
```

Comment:
Could we replace the `.unwrap()` with an early return of `None`?
It would be best to simply not show the module than to panic.

Reply:
yeah, `.unwrap()` was quite rude ;-)

How about the updated version (which is using `.unwrap_or_default()`)?
I also added a couple of additional tests:
- if the current working dir isn't recognized as an hg repo, the module shouldn't produce any output at all
- if it is an hg repo, but the module fails to retrieve a bookmark or branch name (e.g. because it fails to execute hg, not unlikely if it is not installed), it defaults to "(no branch)" (which let the user know that, even if an hg repo has been detected, something went wrong and so the branch name is missing)

How that sounds/looks to you?

Reply:
This is certainly better, but I think the more semantic solution here is to use `None` instead of an empty string.

I think something along these lines would 
```rust
let get_branch_name = |tmpl| get_hg_log_template(tmpl, context, &config);

let mut branch_name = get_branch_name("{activebookmark}");

if branch_name.is_none() {
    branch_name = get_branch_name("{branch}");
}

if branch_name.is_none() {
    branch_name = "(no branch)".to_string();
}
```

Perhaps a simpler way of addressing this without the `if`s would be to use some `Option` methods:
```rust
// I haven't ran this code, so I don't know if it compiles
let get_branch_name = |tmpl| get_hg_log_template(tmpl, context, &config);

let mut branch_name = get_branch_name("{activebookmark}")
	.or_else(|| get_branch_name("{branch}"))
	.unwrap_or("(no branch)".to_string());
```

Reply:
yeah, you are right, I just had to turn an empty string output into None on the `get_hg_log_template` side  (because the '{activebookmark}' output could be an empty string and it would be good to fallback to the branch name in that case)

Thanks! This is so much nicer.